pkg/appgw/requestroutingrules_test.go

```diff
@@ -935,6 +935,50 @@ var _ = Describe("Test routing rules generations", func() {
 		It("has rewrite rule set", func() {
 			Expect(requestRoutingRules[0].RewriteRuleSet).To(Equal(expectedRewriteRuleSet))
 		})
+	})
+
+	Context("test ingress cors annotation", func() {
+		configBuilder := newConfigBuilderFixture(nil)
+		service := tests.NewServiceFixture(*tests.NewServicePortsFixture()...)
+		ingress := tests.NewIngressTestFixtureBasic(tests.Namespace, "random", false)
+		corsAnnotation := "appgw-cors-enable"
+		ingress.Annotations[annotations.EnableCORSKey] = corsAnnotation
+
+		cbCtx := &ConfigBuilderContext{
+			IngressList:           []*networking.Ingress{ingress},
+			ServiceList:           []*v1.Service{service},
+			DefaultAddressPoolID:  to.StringPtr("xx"),
+			DefaultHTTPSettingsID: to.StringPtr("yy"),
+		}
+
+		_ = configBuilder.BackendHTTPSettingsCollection(cbCtx)
+		_ = configBuilder.BackendAddressPools(cbCtx)
+		_ = configBuilder.Listeners(cbCtx)
 
+		pathMap := configBuilder.getPathMaps(cbCtx)
+
+		rule := &ingress.Spec.Rules[0]
+		listenerID := generateListenerID(ingress, rule, n.ApplicationGatewayProtocolHTTP, nil, false)
+		var ruleSets *[]n.ApplicationGatewayRewriteRuleSet = configBuilder.appGw.ApplicationGatewayPropertiesFormat.RewriteRuleSets
+		var corsEnabled bool
+		for i := 0; i < len(*ruleSets); i++ {
```

Comment:
This line is throwing nil exception in the test
![image](https://user-images.githubusercontent.com/5294363/154135961-7cf9909e-db90-46c2-9cb6-54dd530b6a41.png)